### PR TITLE
ub1404 development and modification instructions 

### PR DESCRIPTION
**Pull request by deargle**
See #421 and #416

I switched back to using downloaded versions of the chef vendor cookbooks from the chef marketplace, just because maybe easier to get setup for development for new users. In future, if the chef vendor cookbook versions get bumped often, perhaps I'll revisit and PR a git submodule approach. Chef does have tools to download cookbook dependencies (`knife`, `berks`), but using them requires downloading the beefy chef development kit, which otherwise isn't necessary if only using chef for vagrant provisioning.

I also moved the Vagrantfile to `/chef/dev/ub1404`, per your review request.

### CODE REVIEW

**Comment by Sliim** [comment]
Berkshelf could be used to manage cookbook dependencies instead of copying the whole cookbooks code in this repo...
https://docs.chef.io/berkshelf.html

**Comment by deargle** [comment]
Right, but using berkshelf would require installing the chefdk. Which I
guess isnt a dealbreaker. Adds time to the packer build and testing cycle.

But on the other hand, metasploitable3 already has vendor chef cookbooks
inside of its repo, before this PR.

File: chef/dev/ub1404/,gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.vagrant/
```
[the remark this answers is not in the document]
Comment (deargle) [reply chef/dev/ub1404/,gitignore]: D'oh, it started with a comma anyhow, and it was already included in the root .gitignore. Fixed.

**Comment by deargle** [comment]
As for squashing commits, are you okay with just squashing _everything_ in this PR down to one commit before the final PR? Are you ready for that?

**Comment by jmartin-r7** [comment]
Squash to a single commit is fine, I have tested this and passes muster so far.  If you squash and force push to your branch it will update this PR.

**Comment by deargle** [comment]
The squashenning has come to pass

**Comment by deargle** [comment]
Edited the commit message

**Comment by deargle** [comment]
Shoot, I might have messed up your commit just now with my commit message edit, mb if so

**Pull request merged**